### PR TITLE
added PASCAL VOC dataset

### PR DESCRIPTION
**Pull request by Jiaolong**
Tested on VOC2012 only.

### CODE REVIEW

File: modules/datasets/include/opencv2/datasets/or_pascal.hpp [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+#include <opencv2/core.hpp>
+
+namespace cv
+{
+namespace datasets
+{
+
+//! @addtogroup datasets_or
+//! @{
+struct PascalPart: public Object
```
Comment (bmagyar) [inline modules/datasets/include/opencv2/datasets/or_pascal.hpp]: The indentation for the structs here still seems to be tab-based.
Please exchange them for 4 spaces as of: http://code.opencv.org/projects/opencv/wiki/Coding_Style_Guide#File-Structure

File: modules/datasets/include/opencv2/datasets/or_pascal.hpp [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+//
+//  By downloading, copying, installing or using the software you agree to this license.
+//  If you do not agree to this license, do not download, install,
+//  copy or use the software.
+//
+//
+//                           License Agreement
+//                For Open Source Computer Vision Library
+//
+// Copyright (C) 2014, Itseez Inc, all rights reserved.
```
Comment (bmagyar) [inline modules/datasets/include/opencv2/datasets/or_pascal.hpp]: 2015 :) and there are a few more instances of the same

File: modules/datasets/samples/or_pascal.cpp [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+    printf("\ntrain:\nsize: %u", (unsigned int)train.size());
+    example = static_cast<OR_pascalObj *>(train[0].get());
+    printf("\nfirst image: \n%s", example->filename.c_str());
+    
+    printf("\nsize:");
+    printf("\n - width: %d", example->width);
+    printf("\n - height: %d", example->height);
+    printf("\n - depth: %d", example->depth);
+
+    for (unsigned int i = 0; i < example->objects.size(); i++)
```
Comment (bmagyar) [inline modules/datasets/samples/or_pascal.cpp]: I found it much more error prone to use size_t in for loops when operating over elements of std::vector whereever possible. It saves some trouble when working on both 32bit/64bit compilations.

File: modules/datasets/include/opencv2/datasets/or_pascal.hpp (hunk 1 above)
Comment (Jiaolong) [inline modules/datasets/include/opencv2/datasets/or_pascal.hpp]: Thanks! I have corrected them.

File: modules/datasets/include/opencv2/datasets/or_pascal.hpp (hunk 2 above)
Comment (Jiaolong) [inline modules/datasets/include/opencv2/datasets/or_pascal.hpp]: Done!

File: modules/datasets/samples/or_pascal.cpp (hunk 3 above)
Comment (Jiaolong) [inline modules/datasets/samples/or_pascal.cpp]: So how to change this line?

Comment (bmagyar) [inline modules/datasets/samples/or_pascal.cpp]: size_t i = 0

Comment (Jiaolong) [inline modules/datasets/samples/or_pascal.cpp]: It seem using `size_t i = 0` causes warnings:
http://pullrequest.opencv.org/buildbot/builders/precommit-contrib_linux64/builds/815/steps/compile%20release/logs/warnings%20%281%29
`or_pascal.cpp:87:32: warning: format ‘%d’ expects argument of type ‘int’, but argument 2 has type ‘size_t {aka long unsigned int}’ [-Wformat=]`

**Comment by bmagyar** [comment]
:+1:

**Pull request merged**